### ISSUE TITLE
On the failure of message sending

### ISSUE THREAD

**Issue by YueLiXing** [novastone-media/MQTT-Client-Framework#449]
Based on the current mechanism, we can only know that the message is sent successfully, but the callback of the failed message failed.
Please add

**Comment by jcavar** [comment]
There is async API:

`- (UInt16)publishData:(NSData *)data onTopic:(NSString *)topic retain:(BOOL)retainFlag qos:(MQTTQosLevel)qos publishHandler:(MQTTPublishHandler)publishHandler;`

and publish handler exposes error if it happens.

Is that what you need?

**Comment by YueLiXing** [comment]
I am sorry.
I saw the API in the `MQTTSessionManager`,I've noticed this in the `MQTTSession`.
If you can extract this API to `MQTTSessionManager`, easy for use.

**Comment by jcavar** [comment]
Yes, I agree, we have problem with inconsistent APIs.
I am however not sure if we just want to duplicate this in manager, because that might lead to bloated manager.

I think manager could easily be extracted as separate project and we can consider its design separately.

**Comment by YueLiXing** [comment]
Yes, when I first used this library, I noticed that there were two folders, `Manager` and `Min`. Then I found that I only needed API in `Manager`，This leads me to not find such powerful APi that is so easy to use.I believe that more than one person made this mistake.

If the manager will be edesigned,what a good news！
